src/rendering/swrenderer/textures/warptexture.cpp

```diff
@@ -60,8 +61,9 @@ const uint32_t *FWarpTexture::GetPixelsBgra()
 	if (time != GenTime[2])
 	{
 		auto otherpix = FSoftwareTexture::GetPixelsBgra();
-		WarpedPixelsRgba.Resize(GetWidth() * GetHeight());
-		WarpBuffer(WarpedPixelsRgba.Data(), otherpix, GetWidth(), GetHeight(), WidthOffsetMultiplier, HeightOffsetMultiplier, time, mTexture->shaderspeed, bWarped);
+		WarpedPixelsRgba.Resize(GetWidth() * GetHeight() * gl_texture_hqresizemult * gl_texture_hqresizemult * 4 / 3 + 1);
```

Comment:
That `+ 1` looks a bit suspicious. Doesn’t lower mop level require some alignment, like 4 or 16 bytes?

Reply:
I put it in there in order to ensure that rounding errors would not lower the amount of space required for mipmapped textures. After some calculation I realized that for every level of mipmap, that the required amount for mipmaps will *never* exceed 1/3rd of the total texture size, even if mipmaps are continually generated off of the previous one, but that wholly depends on how the mipmaps were stored in the TArray.

As for the alignment - I was not too sure about that, myself. However, I do notice that this allocation can be completely arbitrary, i.e. if a texture is 3x13, it supposedly should work. If I am wrong about this, feel free to correct me, but I suspect there's some intermediary code occurring with the resize that makes this safe.

Reply:
Actually, I have no idea how mipmaps work in software renderer. If they don't require any alignment then `+ 1` is probably OK.